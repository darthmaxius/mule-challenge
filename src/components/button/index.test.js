import { render, fireEvent } from '@testing-library/react';
import Button from '.';

describe('Button', () => {
  it('Render', () => {
    const { asFragment } = render(<Button className="test">Button</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Click', () => {
    const handler = jest.fn();
    const button = render(
      <Button className="test" onClickHandler={handler}>
        Button
      </Button>,
    ).getByLabelText('panel-button');

    fireEvent.click(button);

    expect(handler).toBeCalled();
  });
});
