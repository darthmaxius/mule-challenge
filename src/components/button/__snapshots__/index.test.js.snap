// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Render 1`] = `
<DocumentFragment>
  <button
    aria-label="panel-button"
    class="test"
  >
    Button
  </button>
</DocumentFragment>
`;
